### ISSUE TITLE
The question of the loss during training. The loss exploded and became nan

### ISSUE THREAD

**Issue by liuyanzhi1214** [open-mmlab/mmdetection#4299]
The loss exploded and became nan
2020-12-14 11:16:28,620 - mmdet - INFO - Epoch [1][652/2936]	lr: 2.000e-02, eta: 1 day, 13:24:07, time: 0.451, data_time: 0.033, memory: 6365, loss_rpn_cls: 0.2981, loss_rpn_bbox: 1.1203, loss_cls: 0.0181, acc: 99.8047, loss_bbox: 0.0023, loss: 1.4387
2020-12-14 11:16:29,094 - mmdet - INFO - Epoch [1][653/2936]	lr: 2.000e-02, eta: 1 day, 13:24:13, time: 0.474, data_time: 0.033, memory: 6365, loss_rpn_cls: 2.5398, loss_rpn_bbox: 0.3872, loss_cls: 0.0379, acc: 99.6094, loss_bbox: 0.0075, loss: 2.9724
2020-12-14 11:16:29,544 - mmdet - INFO - Epoch [1][654/2936]	lr: 2.000e-02, eta: 1 day, 13:24:08, time: 0.451, data_time: 0.033, memory: 6365, loss_rpn_cls: 23.8343, loss_rpn_bbox: 188.3622, loss_cls: 0.0415, acc: 99.0234, loss_bbox: 0.0064, loss: 212.2444
2020-12-14 11:16:29,988 - mmdet - INFO - Epoch [1][655/2936]	lr: 2.000e-02, eta: 1 day, 13:24:01, time: 0.444, data_time: 0.032, memory: 6365, loss_rpn_cls: 37.6083, loss_rpn_bbox: 3.5036, loss_cls: 0.0810, acc: 99.8047, loss_bbox: 0.0041, loss: 41.1970
2020-12-14 11:16:30,441 - mmdet - INFO - Epoch [1][656/2936]	lr: 2.000e-02, eta: 1 day, 13:23:58, time: 0.453, data_time: 0.033, memory: 6365, loss_rpn_cls: 2678.1003, loss_rpn_bbox: 1743.3223, loss_cls: 0.0791, acc: 98.8281, loss_bbox: 0.0286, loss: 4421.5308
2020-12-14 11:16:30,890 - mmdet - INFO - Epoch [1][657/2936]	lr: 2.000e-02, eta: 1 day, 13:23:52, time: 0.449, data_time: 0.034, memory: 6365, loss_rpn_cls: 40165136.0000, loss_rpn_bbox: 1099991.7500, loss_cls: 0.1350, acc: 99.2188, loss_bbox: 0.0229, loss: 41265128.0000
2020-12-14 11:16:31,356 - mmdet - INFO - Epoch [1][658/2936]	lr: 2.000e-02, eta: 1 day, 13:23:55, time: 0.466, data_time: 0.033, memory: 6365, loss_rpn_cls: 5556706916652023808.0000, loss_rpn_bbox: 1248356365118734336.0000, loss_cls: 0.0614, acc: 99.8047, loss_bbox: 0.0112, loss: 6805063281770758144.0000
2020-12-14 11:16:31,802 - mmdet - INFO - Epoch [1][659/2936]	lr: 2.000e-02, eta: 1 day, 13:23:48, time: 0.446, data_time: 0.033, memory: 6365, loss_rpn_cls: 0.6435, loss_rpn_bbox: 0.0156, loss_cls: nan, acc: 0.1953, loss_bbox: nan, loss: nan
2020-12-14 11:16:32,242 - mmdet - INFO - Epoch [1][660/2936]	lr: 2.000e-02, eta: 1 day, 13:23:39, time: 0.440, data_time: 0.033, memory: 6365, loss_rpn_cls: 0.6589, loss_rpn_bbox: 0.0118, loss_cls: nan, acc: 0.1953, loss_bbox: nan, loss: nan
2020-12-14 11:16:32,688 - mmdet - INFO - Epoch [1][661/2936]	lr: 2.000e-02, eta: 1 day, 13:23:33, time: 0.446, data_time: 0.033, memory: 6365, loss_rpn_cls: 0.6461, loss_rpn_bbox: 0.0531, loss_cls: nan, acc: 0.3906, loss_bbox: nan, loss: nan
2020-12-14 11:16:33,130 - mmdet - INFO - Epoch [1][662/2936]	lr: 2.000e-02, eta: 1 day, 13:23:24, time: 0.442, data_time: 0.033, memory: 6365, loss_rpn_cls: 0.6388, loss_rpn_bbox: 0.0033, loss_cls: nan, acc: 0.1953, loss_bbox: nan, loss: nan
2020-12-14 11:16:33,596 - mmdet - INFO - Epoch [1][663/2936]	lr: 2.000e-02, eta: 1 day, 13:23:26, time: 0.465, data_time: 0.033, memory: 6365, loss_rpn_cls: 0.6373, loss_rpn_bbox: 0.0119, loss_cls: nan, acc: 0.0000, loss_bbox: nan, loss: nan
2020-12-14 11:16:34,040 - mmdet - INFO - Epoch [1][664/2936]	lr: 2.000e-02, eta: 1 day, 13:23:20, time: 0.445, data_time: 0.033, memory: 6365, loss_rpn_cls: 0.6399, loss_rpn_bbox: 0.0924, loss_cls: nan, acc: 0.9766, loss_bbox: nan, loss: nan

This is  my configs:
sys.platform: linux
Python: 3.7.9 (default, Aug 31 2020, 12:42:55) [GCC 7.3.0]
CUDA available: True
GPU 0: GeForce RTX 2080 Ti
CUDA_HOME: :/usr/local/cuda-10.1:
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.5.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.6.0a0+82fd1c8
OpenCV: 4.4.0
MMCV: 1.2.1
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 10.1
MMDetection: 2.7.0+
------------------------------------------------------------

2020-12-14 11:11:25,147 - mmdet - INFO - Distributed training: False
2020-12-14 11:11:25,510 - mmdet - INFO - Config:
model = dict(
    type='FasterRCNN',
    pretrained='open-mmlab://resnest50',
    backbone=dict(
        type='ResNeSt',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=False,
        style='pytorch',
        stem_channels=64,
        radix=2,
        reduction_factor=4,
        avg_down_stride=True),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        num_outs=5),
    rpn_head=dict(
        type='RPNHead',
        in_channels=256,
        feat_channels=256,
        anchor_generator=dict(
            type='AnchorGenerator',
            scales=[8],
            ratios=[0.5, 1.0, 2.0],
            strides=[4, 8, 16, 32, 64]),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoder',
            target_means=[0.0, 0.0, 0.0, 0.0],
            target_stds=[1.0, 1.0, 1.0, 1.0]),
        loss_cls=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0),
        loss_bbox=dict(type='L1Loss', loss_weight=1.0)),
    roi_head=dict(
        type='StandardRoIHead',
        bbox_roi_extractor=dict(
            type='SingleRoIExtractor',
            roi_layer=dict(type='RoIAlign', output_size=7, sampling_ratio=0),
            out_channels=256,
            featmap_strides=[4, 8, 16, 32]),
        bbox_head=dict(
            type='Shared4Conv1FCBBoxHead',
            in_channels=256,
            fc_out_channels=1024,
            roi_feat_size=7,
            num_classes=2,
            bbox_coder=dict(
                type='DeltaXYWHBBoxCoder',
                target_means=[0.0, 0.0, 0.0, 0.0],
                target_stds=[0.1, 0.1, 0.2, 0.2]),
            reg_class_agnostic=False,
            loss_cls=dict(
                type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0),
            loss_bbox=dict(type='L1Loss', loss_weight=1.0),
            conv_out_channels=256,
            norm_cfg=dict(type='BN', requires_grad=True))))
train_cfg = dict(
    rpn=dict(
        assigner=dict(
            type='MaxIoUAssigner',
            pos_iou_thr=0.7,
            neg_iou_thr=0.3,
            min_pos_iou=0.3,
            match_low_quality=True,
            ignore_iof_thr=-1),
        sampler=dict(
            type='RandomSampler',
            num=256,
            pos_fraction=0.5,
            neg_pos_ub=-1,
            add_gt_as_proposals=False),
        allowed_border=-1,
        pos_weight=-1,
        debug=False),
    rpn_proposal=dict(
        nms_across_levels=False,
        nms_pre=2000,
        nms_post=1000,
        max_num=1000,
        nms_thr=0.7,
        min_bbox_size=0),
    rcnn=dict(
        assigner=dict(
            type='MaxIoUAssigner',
            pos_iou_thr=0.5,
            neg_iou_thr=0.5,
            min_pos_iou=0.5,
            match_low_quality=False,
            ignore_iof_thr=-1),
        sampler=dict(
            type='RandomSampler',
            num=512,
            pos_fraction=0.25,
            neg_pos_ub=-1,
            add_gt_as_proposals=True),
        pos_weight=-1,
        debug=False))
test_cfg = dict(
    rpn=dict(
        nms_across_levels=False,
        nms_pre=1000,
        nms_post=1000,
        max_num=1000,
        nms_thr=0.7,
        min_bbox_size=0),
    rcnn=dict(
        score_thr=0.05,
        nms=dict(type='nms', iou_threshold=0.5),
        max_per_img=100))
dataset_type = 'VOCDataset'
data_root = 'data/VOCdevkit/'
img_norm_cfg = dict(
    mean=[123.68, 116.779, 103.939], std=[58.393, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='LoadAnnotations',
        with_bbox=True,
        with_mask=False,
        poly2mask=False),
    dict(
        type='Resize',
        img_scale=(1728, 1728),
        multiscale_mode='range',
        keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(
        type='Normalize',
        mean=[123.68, 116.779, 103.939],
        std=[58.393, 57.12, 57.375],
        to_rgb=True),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1728, 1728),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[123.68, 116.779, 103.939],
                std=[58.393, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=1,
    workers_per_gpu=1,
    train=dict(
        type='RepeatDataset',
        times=1,
        dataset=dict(
            type='VOCDataset',
            ann_file='data/VOCdevkit/VOC2007/ImageSets/Main/train.txt',
            img_prefix='data/VOCdevkit/VOC2007/',
            pipeline=[
                dict(type='LoadImageFromFile'),
                dict(type='LoadAnnotations', with_bbox=True),
                dict(type='Resize', img_scale=(1728, 1728), keep_ratio=True),
                dict(type='RandomFlip', flip_ratio=0.5),
                dict(
                    type='Normalize',
                    mean=[123.675, 116.28, 103.53],
                    std=[58.395, 57.12, 57.375],
                    to_rgb=True),
                dict(type='Pad', size_divisor=32),
                dict(type='DefaultFormatBundle'),
                dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
            ]),
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='LoadAnnotations',
                with_bbox=True,
                with_mask=False,
                poly2mask=False),
            dict(
                type='Resize',
                img_scale=(1728, 1728),
                multiscale_mode='range',
                keep_ratio=True),
            dict(type='RandomFlip', flip_ratio=0.5),
            dict(
                type='Normalize',
                mean=[123.68, 116.779, 103.939],
                std=[58.393, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
        ]),
    val=dict(
        type='VOCDataset',
        ann_file='data/VOCdevkit/VOC2007/ImageSets/Main/val.txt',
        img_prefix='data/VOCdevkit/VOC2007/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1728, 1728),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.68, 116.779, 103.939],
                        std=[58.393, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='VOCDataset',
        ann_file='data/VOCdevkit/VOC2007/ImageSets/Main/test.txt',
        img_prefix='data/VOCdevkit/VOC2007/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1728, 1728),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.68, 116.779, 103.939],
                        std=[58.393, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
evaluation = dict(interval=1, metric='mAP')
optimizer = dict(type='SGD', lr=0.02, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=0.001,
    step=[8, 11])
total_epochs = 100
checkpoint_config = dict(interval=1)
log_config = dict(interval=1, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
norm_cfg = dict(type='BN', requires_grad=True)
work_dir = './work_dirs/faster_rcnn_s50_fpn_syncbn-backbone+head_mstrain-range_1x_coco'
gpu_ids = [0]

**Comment by liuyanzhi1214** [comment]
oh，I have solved this problem.I only have one GPU, the lr should be set to 0.02/8 。Because when you have 8 GPUs, lr is set to 0.02
